okhttp/src/main/java/okhttp3/internal/cache/DiskLruCache.kt

```diff
@@ -386,664 +365,642 @@ public final class DiskLruCache implements Closeable, Flushable {
+    val journalWriter = this.journalWriter!!
+    redundantOpCount++
+    entry.currentEditor = null
+    if (entry.readable || success) {
+      entry.readable = true
+      journalWriter.writeUtf8(CLEAN).writeByte(' '.toInt())
+      journalWriter.writeUtf8(entry.key)
+      entry.writeLengths(journalWriter)
+      journalWriter.writeByte('\n'.toInt())
       if (success) {
-        entry.sequenceNumber = nextSequenceNumber++;
+        entry.sequenceNumber = nextSequenceNumber++
       }
     } else {
-      lruEntries.remove(entry.key);
-      journalWriter.writeUtf8(REMOVE).writeByte(' ');
-      journalWriter.writeUtf8(entry.key);
-      journalWriter.writeByte('\n');
+      lruEntries.remove(entry.key)
+      journalWriter.writeUtf8(REMOVE).writeByte(' '.toInt())
+      journalWriter.writeUtf8(entry.key)
+      journalWriter.writeByte('\n'.toInt())
     }
-    journalWriter.flush();
+    journalWriter.flush()
 
     if (size > maxSize || journalRebuildRequired()) {
-      executor.execute(cleanupRunnable);
+      executor.execute(cleanupRunnable)
     }
   }
 
   /**
    * We only rebuild the journal when it will halve the size of the journal and eliminate at least
    * 2000 ops.
    */
-  boolean journalRebuildRequired() {
-    final int redundantOpCompactThreshold = 2000;
-    return redundantOpCount >= redundantOpCompactThreshold
-        && redundantOpCount >= lruEntries.size();
+  private fun journalRebuildRequired(): Boolean {
+    val redundantOpCompactThreshold = 2000
+    return redundantOpCount >= redundantOpCompactThreshold &&
+        redundantOpCount >= lruEntries.size
   }
 
   /**
-   * Drops the entry for {@code key} if it exists and can be removed. If the entry for {@code key}
-   * is currently being edited, that edit will complete normally but its value will not be stored.
+   * Drops the entry for [key] if it exists and can be removed. If the entry for [key] is currently
+   * being edited, that edit will complete normally but its value will not be stored.
    *
    * @return true if an entry was removed.
    */
-  public synchronized boolean remove(String key) throws IOException {
-    initialize();
-
-    checkNotClosed();
-    validateKey(key);
-    Entry entry = lruEntries.get(key);
-    if (entry == null) return false;
-    boolean removed = removeEntry(entry);
-    if (removed && size <= maxSize) mostRecentTrimFailed = false;
-    return removed;
+  @Synchronized @Throws(IOException::class)
+  fun remove(key: String): Boolean {
+    initialize()
+
+    checkNotClosed()
+    validateKey(key)
+    val entry = lruEntries[key] ?: return false
+    val removed = removeEntry(entry)
+    if (removed && size <= maxSize) mostRecentTrimFailed = false
+    return removed
   }
 
-  boolean removeEntry(Entry entry) throws IOException {
-    if (entry.currentEditor != null) {
-      entry.currentEditor.detach(); // Prevent the edit from completing normally.
-    }
+  @Throws(IOException::class)
```

Comment:
For internal classes, a lot of the throws can be presumably dropped.  Left just on public APIs.

Reply:
I want to keep them as long as we’re supporting Java callers (ie. forever). Otherwise it’s very difficult to know when an `@Throws` is necessary.